### PR TITLE
WINC-723: Pick up kubernetes 1.23 rebase changes

### PR DESCRIPTION
**Pull request by sebsoto**
Includes:
* Updates to kubelet and kube-proxy submodules
* Updates go.mod and vendor to pick up k8s 1.23 rebases

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,9 +5,9 @@ go 1.17
 require (
 	github.com/apparentlymart/go-cidr v1.1.0
 	github.com/aws/aws-sdk-go v1.38.23
-	github.com/go-logr/logr v0.4.0
-	github.com/openshift/api v0.0.0-20211103131146-a0807454a876
-	github.com/openshift/client-go v0.0.0-20211025111749-96ca2abfc56c
+	github.com/go-logr/logr v1.2.0
+	github.com/openshift/api v0.0.0-20211209135129-c58d9f695577
+	github.com/openshift/client-go v0.0.0-20211209144617-7385dd6338e3
 	github.com/openshift/library-go v0.0.0-20210811133500-5e31383de2a7
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply go.mod:11]: There were no changes to library-go or openshift/api yet, I may have to rebase and pick up changes to both of those.

[the remark this answers is not in the document]
Comment (sebsoto) [reply go.mod:11]: openshift/api was updated only as a result of updating client-go

**Review by selansen — APPROVED** [review]
/lgtm

**Comment by sebsoto** [comment]
@aravindhp I dropped the kubernetes 1.23 server requirements from this PR. Will open a new PR including those changes when they are relevant

**Review by aravindhp — APPROVED** [review]
/approve

@sebsoto please open a manual backport PR for `release-4.10`

**Pull request merged**